### DISCUSSION TITLE
What exactly is "Asset Discovery"?

### DISCUSSION THREAD

**Discussion by Naoto-Ida** [percy/cli#1289] [Support] [answered]
Hi Community, we we're trying to figure out some issues with Chromium not launching in one of our self hosted GH runners.

After looking through the docs and trying to figure out what requires Chromium (as we understand Percy to be a service provided by Browserstack, therefore not requiring a browser locally), we've found many mentions of "Asset Discovery".

Could somebody enlighten me on what Asset Discovery actually means? There's no mention of it other than just the phrase, and this may help us understand why it needs Chromium.

For context, we are trying to use [@percy/storybook](https://github.com/percy/percy-storybook).

**Comment by Browseutkarsh** [comment, marked as answer]
Hi,

In the context of @percy/storybook and Chromium, "Asset Discovery" refers to the process of identifying and capturing screenshots of your Storybook components or UI elements. Asset Discovery is a crucial step in visual testing, where you compare visual snapshots of your UI components to identify any visual regressions.

If you're experiencing issues with Chromium not launching in your self-hosted GitHub (GH) runner, it could be related to the configuration of the runner environment or the availability of dependencies required by Chromium which you may want to check.

If you still face issue you can also raise a support ticket by sending an email to [support@browserstack.com](mailto:support@browserstack.com) or via Contact Us(https://www.browserstack.com/contact).

Reference document
- https://docs.percy.io/docs/debugging-sdks#asset-discovery
- https://docs.percy.io/docs/storybook-tutorial

**Comment by Naoto-Ida** [comment]
Thank you!

So just to clarify, this cannot run within the Browserstack infrastructure? Are there plans to be able to upload the static storybook files and process them there?
